java/daikon/dcomp/DCInstrument.java

```diff
@@ -1837,9 +1869,68 @@ void add_exit(MethodGen mg, MethodInfo mi, int method_info_index) {
   }
 
   /**
-   * Discards primitive tags for each primitive argument to a non-instrumented method and adds a tag
-   * for a primitive return value. Ensures that the tag stack is correct for non-instrumented
-   * methods.
+   * Return the interface class containing the implementation of the target method. The interfaces
+   * of {@code startClass} are recursively searched.
+   *
+   * @param startClass the JavaClass whose interfaces are to be searched
+   * @param method_name target method to search for
+   * @param arg_types the target method's argument types
+   * @return name of interface class containing target method, or null if not found
+   */
+  @Nullable @ClassGetName String findTargetInterface(JavaClass startClass, String method_name, Type[] arg_types) {
+
+    if (debugFindTargetInterface) {
+      System.out.println("searching interfaces of: " + startClass.getClassName());
+    }
+    for (String interfaceNameString : startClass.getInterfaceNames()) {
+      @SuppressWarnings("signature:assignment")
+      @ClassGetName String interfaceName = interfaceNameString;
+      if (debugFindTargetInterface) {
+        System.out.println("interface: " + interfaceName);
+      }
+      JavaClass ji;
+      try {
+        ji = getJavaClass(interfaceName);
+      } catch (Throwable e) {
+        throw new Error(String.format("Unable to load class: %s", interfaceName), e);
+      }
+      for (Method jm : ji.getMethods()) {
+        if (debugFindTargetInterface) {
+          System.out.println("  " + jm.getName() + Arrays.toString(jm.getArgumentTypes()));
+        }
+        if (jm.getName().equals(method_name) && Arrays.equals(jm.getArgumentTypes(), arg_types)) {
+          // We have a match.
+          return interfaceName;
+        }
+      }
+      // no match found; does this interface extend other interfaces?
+      @ClassGetName String found = findTargetInterface(ji, method_name, arg_types);
+      if (found != null) {
+        // We have a match.
+        return found;
+      }
+    }
+    // nothing found
+    return null;
+  }
+
+  /**
+   * Process an Invoke instruction. There are three cases:
+   *
+   * <ul>
+   *   <li>convert calls to Object.equals to calls to dcomp_equals or dcomp_super_equals
+   *   <li>convert calls to Object.clone to calls to dcomp_clone or dcomp_super_clone
+   *   <li>otherwise, determine whether the target of the invoke is instrumented or not (this is the
+   *       {@code callee_instrumented} variable)
+   *       <ul>
+   *         <li>If the target method is not instrumented, generate code to discard a primitive tag
```

Comment:
"discard a primitive tag" sounds scary.  Is the information lost?  Why is it OK to lose that information?